Core/Filesystem/nand_core.c

```diff
@@ -0,0 +1,366 @@
+/*
+ * Copyright (C) 2022  University of Alberta
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+/*
+ * nand_core.c
+ *
+ *  Created on: Jun 14, 2022
+ *      Author: Robert Taylor
+ */\
+#include <stdint.h>
+#include <string.h>
+#include "nandfs.h"
+#include "nand_core.h"
+#include "nand_m79a_lld.h"
+#include "nand_types.h"
+#include "nand_errno.h"
+
+static int lowest_inode = 0;
+static int highest_inode = 0;
+
+#define UINT32_MAX 0xFFFFFFFF;
+
+/*
+ * Finds the first unformatted block
+ * @Return:
+ * 0, success, blank space found
+ * -1, failed. Could be I/O or full device, No distinction made
```

Comment:
If we receive -1, I presume we're going off to the second NAND chip? Is there provisions for if both chips nack or are full? If they both are full I think we can begin overwriting the oldest images

Reply:
We haven't yet decided how we want to treat the second chip, and the low level driver doesn't support it, so I just didn't consider the second chip at all here